src/shims/fs.rs

```diff
@@ -257,6 +263,214 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.try_unwrap_io_result(result)
     }
 
+    fn statx(
+        &mut self,
+        dirfd_op: OpTy<'tcx, Tag>,
+        pathname_op: OpTy<'tcx, Tag>,
+        flags_op: OpTy<'tcx, Tag>,
+        _mask_op: OpTy<'tcx, Tag>,
+        statxbuf_op: OpTy<'tcx, Tag>
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        this.check_no_isolation("statx")?;
+
+        let statxbuf_scalar = this.read_scalar(statxbuf_op)?.not_undef()?;
```

Comment:
Same with these variables. I don't think we should be including the type of the variable in the variable name unless the distinction is needed (and then we should probably consider some better naming scheme)

Reply:
yes `statxbuf` was a particularly bad case because all the arguments of `statx` arrive as integers. So handling the `statxbuf_op` operator to transform it into a proper pointer was... messy.